### ISSUE TITLE
Running instance with pre-created NIC with DeviceIndex: 0 fails with API error

### ISSUE THREAD

**Issue by varnastadeus** [aws/aws-sdk-go-v2#1178]
Confirm by changing [ ] to [x] below to ensure that it's a bug:
- [x] I've gone though the [API reference](https://docs.aws.amazon.com/sdk-for-go/v2/api/)
- [x] I've checked [AWS Forums](https://forums.aws.amazon.com) and [StackOverflow](https://stackoverflow.com/questions/tagged/aws-sdk-go) for answers
- [x] I've searched for [previous similar issues](https://github.com/aws/aws-sdk-go-v2/issues) and didn't find any solution
  
**Describe the bug**
Running instance with pre-created network interface and `DeviceIndex` of `0` results in API error: `api error InvalidParameterValue: Each network interface requires a device index.` It seems `0` is treated as an empty value even though it is valid.

In `sdk-v1` it was pointer and we used `DeviceIndex: aws.Int64(0)` which worked perfectly fine.

**Version of AWS SDK for Go?**
`aws-sdk-go-v2/1.2.1`

**Version of Go (`go version`)?**
`go/1.16`

**To Reproduce (observed behavior)**
```go
subnetID := "TODO"
imageID := "TODO"

cfg, err := config.LoadDefaultConfig(context.Background())
if err != nil {
    return err
}

client := ec2.NewFromConfig(cfg)
nic, err := client.CreateNetworkInterface(context.Background(), &ec2.CreateNetworkInterfaceInput{
    SubnetId: &subnetID,
})
if err != nil {
    return err
}

_, err = client.RunInstances(context.Background(), &ec2.RunInstancesInput{
    InstanceType: ec2types.InstanceTypeT2Medium,
    ImageId:      &imageID,
    MaxCount:     1,
    MinCount:     1,
    NetworkInterfaces: []ec2types.InstanceNetworkInterfaceSpecification{
    {
	NetworkInterfaceId: nic.NetworkInterface.NetworkInterfaceId,
	DeviceIndex:        0,
    },
},
})
if err != nil {
    return err
}

return nil
```

**Expected behavior**
The instance should be able to start.

**Comment by jasdel** [comment]
Thanks for reaching out @varnastadeus . This issue is similar to https://github.com/aws/aws-sdk-go-v2/issues/1107 Where the EC2 API model is missing metadata instructing the SDK how it should serialize these operations. We're working with the EC2 service team on how best to address these operations.

**Comment by jasdel** [comment]
This issue has been resolved in #1195 in the latest release tag, `service/ec2/v1.6.0`.  You can update your application to use the latest version with

```sh
go get -d github.com/aws/aws-sdk-go-v2/service/ec2@v1.6.0
```

While this release fixes the client to be able to send all request parameters to the API, a breaking change to the generated code was needed. The parameters like `MaxCount`, `MinCount`, and  `DeviceIndex` are pointer types now, not values. You can use either addresses, or the `aws.Int32` utilities to not need to create a local variable when assigning the value.
